t/deploy.t

```diff
@@ -67,6 +67,7 @@ $schema = OpenQA::Schema::connect_db(mode => 'test', check => 0);
 ensure_schema_is_created_and_empty $schema;
 
 # redeploy DB to older version and check if deployment_check upgrades the DB
+my $old_schema_version = 76;
```

Comment:
I don't think this is long-term sustainable to have a magic number here.

Reply:
I would say that's ok. It makes sense to start this test at a fixed schema version. We might increase this version in the future (similar to the current schema version) when we drop support for old migrations.

Reply:
how about making it a bit smarter with looking into whatever is the oldest db init scripts in the tree and use that?

Reply:
That makes generally sense. I would keep this commit as a short-term-fix at it is and improve the deployment test later on. It is not that simple because I'm currently leaving out 2 of our old migrations which would actually fail. I first need to investigate why they are failing. Maybe we can also phase them out (they are from 2018).

Reply:
I'd say we should have a fixed list of db inits/migrations, i.e. for every new db migration delete the oldest one. However, 2018 isn't *too* old.

Reply:
> i.e. for every new db migration delete the oldest one

Than the duration we keep migrations is determined by the frequency new migrations are added and therefore may vary. I'd prefer keeping the duration fixed and not the amount of migrations. For the user it is more meaningful to know that an e.g. 2 year old instance can still be upgraded than knowing that an instance which is e.g. 10 migrations behind can still be upgraded.

Reply:
sure, we can also combine and ensure at least 20 migrations and at least 2 years are preserved.